### PR TITLE
Debugger elements

### PR DESCRIPTION
**Pull request by Anthony-Eid**
This PR aim's to start the process of cleaning up & improving the maintainability of the debugger's variable rendering and management. 

By creating a new View called `VariableList` and encapsulating the logic written by Remco that handles variable management. Post Alpha of the debugger we will be changing the variable list's internals to be cleaner as well. 

The merging of this PR should be done as a squash commit because there isn't much context that the commit's of the PR add.

### CODE REVIEW

File: crates/debugger_ui/src/debugger_panel_item.rs [hunk 1]
```diff
@@ -37,18 +37,18 @@ pub enum ThreadEntry {
...
+    pub thread_id: u64,
+    variable_list: View<VariableList>,
     focus_handle: FocusHandle,
     stack_frame_list: ListState,
     output_editor: View<Editor>,
-    open_entries: Vec<SharedString>,
-    stack_frame_entries: HashMap<u64, Vec<ThreadEntry>>,
     active_thread_item: ThreadItem,
-    client: Arc<DebugAdapterClient>,
+    pub client: Arc<DebugAdapterClient>,
```
Comment (RemcoSmitsDev) [inline crates/debugger_ui/src/debugger_panel_item.rs:50]: Can you pls use the methods for this, and if we don't have a method create one for it. So it's more inline how Zed does things.

File: crates/debugger_ui/src/debugger_panel_item.rs [hunk 2]
```diff
@@ -37,18 +37,18 @@ pub enum ThreadEntry {
 }
 
 pub struct DebugPanelItem {
-    thread_id: u64,
-    variable_list: ListState,
+    pub thread_id: u64,
```
Comment (RemcoSmitsDev) [inline crates/debugger_ui/src/debugger_panel_item.rs:44]: Can you pls use the methods for this, and if we don't have a method create one for it. So it's more inline how Zed does things.

**Comment by RemcoSmitsDev** [comment]
Can we maybe call the module variable List instead of elements? That is more clear because it only contains variable list code.

**Review by RemcoSmitsDev — CHANGES_REQUESTED** [review]

File: crates/debugger_ui/src/debugger_panel_item.rs (hunk 1 above)
Comment (Anthony-Eid) [reply crates/debugger_ui/src/debugger_panel_item.rs:50]: fixed

File: crates/debugger_ui/src/debugger_panel_item.rs (hunk 2 above)
Comment (Anthony-Eid) [reply crates/debugger_ui/src/debugger_panel_item.rs:44]: fixed

**Pull request merged**